assets/src/components/AssignmentTable.js

```diff
@@ -120,170 +144,271 @@ function AssignmentTable (props) {
     }
   }, [currentWeekRow.current])
 
+  const [assignmentGroupNames, setAssignmentGroupNames] = useState([])
```

Comment:
I see why you declared the state here, but I think the usual convention in React is to declare all state variables near the top of the component. Could you move this line to be with the other `useState` calls above?